tsdb/index/inmem/inmem.go

```diff
@@ -211,6 +210,9 @@ func (i *Index) CreateSeriesIfNotExists(shardID uint64, key, name []byte, tags m
 	// Add the series to the series sketch.
 	i.seriesSketch.Add(key)
 
+	// This series needs to be added to the bitset tracking undeleted series IDs.
+	seriesSet.Add(seriesID)
```

Comment:
I think this might make the need for the `seriesSketch` on L211 unnecessary.  Might want to look at that another time.